clang/lib/CodeGen/CodeGenFunction.cpp

```diff
@@ -2142,14 +2142,10 @@ Address CodeGenFunction::EmitIntelFPGAFieldAnnotations(SourceLocation Location,
   llvm::Value *V = Addr.getPointer();
   llvm::Type *VTy = V->getType();
   llvm::Function *F =
-      CGM.getIntrinsic(llvm::Intrinsic::ptr_annotation, CGM.Int8PtrTy);
-  // FIXME Always emit the cast inst so we can differentiate between
-  // annotation on the first field of a struct and annotation on the struct
-  // itself.
-  if (VTy != CGM.Int8PtrTy)
-    V = Builder.CreateBitCast(V, CGM.Int8PtrTy);
+      CGM.getIntrinsic(llvm::Intrinsic::ptr_annotation, Int8PtrTy);
```

Comment:
`llvm.ptr.annotation` supports pointers with address spaces, so it's better not to cast everything to `i8*`, but keep the address space by casting to `i8 addrspace(N)*`, where `N` matches with an address space of `V->getType()`.

Reply:
Good point, thanks. Done.